src/internal/user_api/requests.rs

```diff
@@ -121,6 +137,86 @@ pub mod user_verify {
     }
 }
 
+pub mod user_get {
+    use super::*;
+
+    #[derive(Deserialize, PartialEq, Debug)]
+    #[serde(rename_all = "camelCase")]
+    pub struct CurrentUserResponse {
+        pub(in crate::internal) current_key_id: u64,
+        pub(in crate::internal) id: String,
+        pub(in crate::internal) status: usize,
+        pub(in crate::internal) segment_id: usize,
+        pub(in crate::internal) user_master_public_key: PublicKey,
+        pub(in crate::internal) user_private_key: EncryptedPrivateKey,
+        pub(in crate::internal) needs_rotation: bool,
+        pub(in crate::internal) groups_needing_rotation: Vec<String>,
+    }
+
+    pub fn get_curr_user(
+        auth: &RequestAuth,
+    ) -> impl Future<Item = CurrentUserResponse, Error = IronOxideErr> + '_ {
+        auth.request.get(
+            "users/current",
+            RequestErrorCode::UserGetCurrent,
+            AuthV2Builder::new(&auth, Utc::now()),
+        )
+    }
+}
+
+/// PUT /users/{userId}/keys/{userKeyId}
+pub mod user_update_private_key {
+    use super::*;
+    use crate::internal::user_api::UserUpdatePrivateKeyResult;
+
+    #[derive(Serialize, Debug)]
+    #[serde(rename_all = "camelCase")]
+    pub struct UserUpdatePrivateKey {
+        user_private_key: EncryptedPrivateKey,
+        augmentation_factor: AugmentationFactor,
```

Comment:
I'm not seeing the Schnorr signature - is it wrapped up in the `EncryptedPrivateKey`?

Reply:
Should user private key rotation use a Schnorr signature? As of now we don't. Since this call is made with a device it doesn't seem necessary to me at first glance.

Reply:
My mistake - the thing we sign with Schnorr is in the device transform key, to make sure that was generated with the updated private key.